### PR TITLE
Add priority and done method.

### PR DESCRIPTION
**Pull request by sianghwee**

### CODE REVIEW

File: src/main/java/seedu/address/commons/core/item/Item.java [hunk 1]
```diff
@@ -240,19 +240,19 @@ public String toString() {
         public ItemBuilder() {}
 
         public ItemBuilder setTask(Task task) {
-            requireNonNull(task);
+            //requireNonNull(task);
```
Comment (lrchema) [inline src/main/java/seedu/address/commons/core/item/Item.java]: I think it's better to delete the code rather than comment it out

Comment (sianghwee) [reply src/main/java/seedu/address/commons/core/item/Item.java]: Ok! However, I choose to comment it out as it is @blimyj code, so should leave it for him to remove it.

File: src/main/java/seedu/address/model/ItemModelManager.java [hunk 2]
```diff
@@ -234,4 +253,72 @@ public void clear() {
...
+    private void togglePriorityMode() {
+        priorityMode = !priorityMode;
+    }
+
+    /**
+     * Enable and disable the priority mode
+     * @return a boolean value. If true, means priority mode is on, else returns false.
+     * @throws IllegalListException if the visualList is not a task list.
+     */
+    public boolean togglePriority() throws IllegalListException {
```
Comment (blimyj) [inline src/main/java/seedu/address/model/ItemModelManager.java]: Do you want to merge this method with togglePriorityMode while retaining the method name togglePriorityMode?
There might be some confusion between togglePriority and changePriority.

File: src/main/java/seedu/address/model/ItemModelManager.java [hunk 3]
```diff
@@ -234,4 +253,72 @@ public void clear() {
...
+        return result;
+    }
+
+    /**
+     * Mark an item with a task as done.
+     * @param index the index of the item to be marked as done.
+     * @return the item that is marked as done.
+     * @throws IllegalListException if the operation is not done on a task list.
+     */
+    public Item done(int index) throws IllegalListException {
```
Comment (blimyj) [inline src/main/java/seedu/address/model/ItemModelManager.java]: Do you want to name this method markComplete instead? So as to maintain consistency with the naming of markComplete method.

File: src/main/java/seedu/address/model/ItemModelManager.java (hunk 2 above)
Comment (sianghwee) [reply src/main/java/seedu/address/model/ItemModelManager.java]: Ok!

File: src/main/java/seedu/address/model/ItemModelManager.java (hunk 3 above)
Comment (sianghwee) [reply src/main/java/seedu/address/model/ItemModelManager.java]: ok that's fine with me

File: src/main/java/seedu/address/model/ItemModelManager.java [hunk 4]
```diff
@@ -234,4 +253,68 @@ public void clear() {
...
+
+        priorityMode = !priorityMode;
+        if (!priorityMode) {
+            sortedTask = null;
+            this.visualList = taskList;
+        } else {
+            sortedTask = new PriorityQueue<>((item1, item2) -> {
+                Task task1 = item1.getTask().get();
+                Task task2 = item2.getTask().get();
+                return task1.getPriority().compareTo(task2.getPriority());
```
Comment (mannggoo) [inline src/main/java/seedu/address/model/ItemModelManager.java]: Great work! But just wondering, I thought getPriority() returns HIGH/MEDIUM/LOW? How is that compared?

Comment (blimyj) [reply src/main/java/seedu/address/model/ItemModelManager.java]: It's compared with the ordinal value of the enum!
https://docs.oracle.com/javase/7/docs/api/java/lang/Enum.html

**Pull request merged**